testsuite/tests/basic-modules/recursive_module_evaluation_errors.ml

```diff
@@ -17,19 +17,19 @@ Error: Cannot safely evaluate the definition of the following cycle
 Line 2, characters 10-20:
 2 | and B:sig val x: int end = struct let x = E.y end
               ^^^^^^^^^^
-  Module "B" defines an unsafe value, "x" .
+  Module "B" defines an unsafe value, "B.x" .
 Line 5, characters 10-20:
 5 | and E:sig val x: int val y:int end = struct let x = D.x let y = 0 end
               ^^^^^^^^^^
-  Module "E" defines an unsafe value, "x" .
+  Module "E" defines an unsafe value, "E.x" .
```

Comment:
Note: I'm not sure about the choice of repeating the toplevel module name in the unsafe value path. I wonder if this is an explicit choice that you made, or a consequence of the current implementation.

Overall my impression is that:
- repeating the toplevel module name is slightly useful in the case of unsafe values within nested submodules
- when the unsafe value is at the toplevel of the module (the common case) this is slightly unhelpful
- all other things being equal, I would have a preference for not changing the current error message in the case where the unsafe value is at the toplevel, if only to avoid testsuite churn, but having things changed is not a big deal either

That is: no strong opinion, but maybe if the modified implementation makes it easy to *not* include the toplevel module name I would be in favor of not including it.